js/components/ps-number-input.js

```diff
@@ -19,15 +19,8 @@ const PSNumberInput = function PSNumberInput(element) {
   let value = input.value;
 
   let initValue = () => {
-    if (
-      input.value.replace('.', '') !== input.value ||
-      input.value.replace(',', '') !== input.value
-    ) {
-      value = parseInt(
-        input.value.replace(',', '') !== input.value
-          ? input.value.replace(',', '.')
-          : input.value
-      );
+    if (input.value.replace('.', '') !== input.value || input.value.replace(',', '') !== input.value) {
+      value = parseInt(input.value.replace(',', '') !== input.value ? input.value.replace(',', '.') : input.value);
     } else {
       value = Number(input.value);
     }
```

Comment:
This can be optimized with a regex:


```suggestion
    const sanitized = input.value.replace(/[.,]/g, '');
    value = (sanitized !== input.value) ? parseInt(sanitized) : Number(input.value);
```

I'm not sure about the `parseInt` vs `Number`. I'm guessing this field always returns a string? In that case we should always use `parseInt`.

That said, I'm not seeing a test for `isNaN`...